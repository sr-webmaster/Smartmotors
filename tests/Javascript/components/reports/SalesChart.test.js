import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '~/components/reports/SalesChart'

const salesData = {
  Statuses: [
    { key: 'cancelled', doc_count: 10 },
    { key: 'lead', doc_count: 20 },
    { key: 'delivered', doc_count: 30 },
    { key: 'prospect', doc_count: 40 },
    { key: 'sold', doc_count: 50 },
    { key: 'quote', doc_count: 60 }
  ],
  AccessoryGross: { value: 100000 },
  FinanceCharges: { value: 110000 },
  FinanceGross: { value: 120000 },
  DealGross: { value: 130000 }
}

const columnsData = {
  data: [
    { label: 'Delivered', red: 10, green: 20, blue: 30 },
    { label: 'Pending', red: 20, green: 30, blue: 40 },
    { label: 'Secured', red: 30, green: 40, blue: 50 },
    { label: 'Incoming', red: 30, green: 40, blue: 50 },
    { label: 'Order Out', red: 30, green: 40, blue: 50 },
    { label: 'Dead', red: 30, green: 40, blue: 50 }
  ],
  sets: [
    { field: 'red', color: 'red' },
    { field: 'green', color: 'green' },
    { field: 'blue', color: 'blue' }
  ]
}

const ringGraphData = {
  data: [
    { label: 'Delivered', value: 10 },
    { label: 'Pending', value: 20 },
    { label: 'Secured', value: 30 },
    { label: 'Incoming', value: 30 },
    { label: 'Order Out', value: 30 },
    { label: 'Dead', value: 30 }
  ]
}

describe('SalesChart', () => {
  let localVue

  describe('Snapshot', () => {
    test('default', () => {
      global.SVGPathElement = jest.fn()
      localVue = createLocalVue()

      const wrapper = shallowMount(Component, {
        propsData: {
          columnsData: columnsData,
          ringGraphData: ringGraphData
        },
        localVue: localVue
      })

      expect(wrapper.element).toMatchSnapshot()
    })
    test('with sales data', () => {
      global.SVGPathElement = jest.fn()
      localVue = createLocalVue()

      const wrapper = shallowMount(Component, {
        propsData: {
          salesData: salesData,
          columnsData: columnsData,
          ringGraphData: ringGraphData
        },
        localVue: localVue
      })

      expect(wrapper.text()).toContain('210')
      expect(wrapper.text()).toContain('50')
      expect(wrapper.text()).toContain('30')
      expect(wrapper.text()).toContain('60')
      expect(wrapper.text()).toContain('$100,000.00')
      expect(wrapper.text()).toContain('$110,000.00')
      expect(wrapper.text()).toContain('$120,000.00')
      expect(wrapper.text()).toContain('$130,000.00')
      expect(wrapper.element).toMatchSnapshot()
    })

    test('loading', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          isLoading: true
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Computed', () => {
    global.SVGPathElement = jest.fn()
    localVue = createLocalVue()

    const wrapper = shallowMount(Component, { localVue })
    wrapper.setData({ salesData })

    expect(wrapper.vm._salesData).toEqual([
      { title: 'Total Deals Written This period', value: 210 },
      { title: 'Total Deals Sold This period', value: 50 },
      { title: 'Total Deals Delivered This period', value: 30 },
      { title: 'Total Deals Quote This period', value: 60 },
      { title: 'Accessory Sales This period', value: 100000, money: true },
      { title: 'Finance Charges This period', value: 110000, money: true },
      { title: 'Finance Gross This period', value: 120000, money: true },
      { title: 'Gross Sales This period', value: 130000, money: true }
    ])
  })
})
