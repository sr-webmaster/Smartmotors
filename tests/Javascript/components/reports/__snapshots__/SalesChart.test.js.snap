// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalesChart Snapshot default 1`] = `
<div
  ringgraphdata="[object Object]"
>
  <v-layout-stub
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Total Deals Written This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          0
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Total Deals Sold This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          0
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Total Deals Delivered This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          0
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Total Deals Quote This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          0
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Accessory Sales This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          $0.00
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Finance Charges This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          $0.00
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Finance Gross This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          $0.00
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Gross Sales This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          $0.00
        
        </div>
      </v-card-stub>
    </v-flex-stub>
  </v-layout-stub>
   
  <v-layout-stub
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      class="[object Object]"
      sm7=""
      tag="div"
    >
      <div
        class="card-title"
      >
        
        Total Sales By Status / Location
      
      </div>
       
      <columns-stub
        chartdata="[object Object]"
      />
    </v-flex-stub>
     
    <v-flex-stub
      class="[object Object]"
      tag="div"
    >
      <div
        class="card-title"
      >
        
        Total Sales by Status
      
      </div>
       
      <ring-graph-stub
        chartdata="[object Object]"
      />
    </v-flex-stub>
  </v-layout-stub>
</div>
`;

exports[`SalesChart Snapshot loading 1`] = `
<v-progress-linear-stub
  active="true"
  buffervalue="100"
  color="primary"
  height="7"
  indeterminate="true"
  value="0"
/>
`;

exports[`SalesChart Snapshot with sales data 1`] = `
<div
  ringgraphdata="[object Object]"
>
  <v-layout-stub
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Total Deals Written This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          210
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Total Deals Sold This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          50
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Total Deals Delivered This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          30
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Total Deals Quote This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          60
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Accessory Sales This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          $100,000.00
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Finance Charges This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          $110,000.00
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Finance Gross This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          $120,000.00
        
        </div>
      </v-card-stub>
    </v-flex-stub>
    <v-flex-stub
      class="[object Object]"
      sm3=""
      tag="div"
    >
      <v-card-stub
        class="px-1 py-3 text-xs-center"
        tag="div"
      >
        <div
          class="card-title ellipsis-text"
        >
          
          Gross Sales This period
        
        </div>
         
        <div
          class="headline pt-1"
        >
          
          $130,000.00
        
        </div>
      </v-card-stub>
    </v-flex-stub>
  </v-layout-stub>
   
  <v-layout-stub
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      class="[object Object]"
      sm7=""
      tag="div"
    >
      <div
        class="card-title"
      >
        
        Total Sales By Status / Location
      
      </div>
       
      <columns-stub
        chartdata="[object Object]"
      />
    </v-flex-stub>
     
    <v-flex-stub
      class="[object Object]"
      tag="div"
    >
      <div
        class="card-title"
      >
        
        Total Sales by Status
      
      </div>
       
      <ring-graph-stub
        chartdata="[object Object]"
      />
    </v-flex-stub>
  </v-layout-stub>
</div>
`;
